azurelinuxagent/common/utils/cryptutil.py

```diff
@@ -128,3 +141,17 @@ def bits_to_bytes(self, bits):
                 index = 7
         return bytes(byte_array)
 
+    def decodeBase64ToFile(self, cacheFile, base64str):
+        with open(cacheFile, "wb") as c:
+            c.write(base64.b64decode(base64str))
+            #c.write(base64.decodestring(base64str))
+    
+    def decryptSecret(self, encryptedPassword, privateKey, cacheFile):
```

Comment:
Why are you passing in a cacheFile?  If this is a temporary file, you should just create it in this method.

Reply:
To avoid possible threading issues if this gets used elsewhere in the future.